buildlog/buildlog_test.go

```diff
@@ -0,0 +1,198 @@
+package buildlog
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	intervalTime = 500
+	timeRegEx    = `[\d]{4}-[\d]{2}-[\d]{2} [\d]{2}:[\d]{2}:[\d]{2}`
+)
+
+var (
+	testInputs = []string{
+		`{"t": 1581662022394, "m": "test 1", "n": 0, "s": "main"}` + "\n",
+		`{"t": 1581662022395, "m": "test 2", "n": 1, "s": "main"}` + "\n",
+	}
+)
+
+type writeCloser struct {
+	io.Writer
+}
+
+func (wc writeCloser) Close() error {
+	return nil
+}
+
+func write(tb testing.TB, filepath string, inputs []string) {
+	tb.Helper()
+
+	for _, input := range inputs {
+		file, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+		if err != nil {
+			tb.Fatal(err)
+		}
+
+		_, err = file.Write([]byte(input))
+		if err != nil {
+			tb.Fatal(err)
+		}
+
+		err = file.Close()
+		if err != nil {
+			tb.Fatal(err)
+		}
+	}
+}
+
+func TestRun(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		tmpFile, err := ioutil.TempFile("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer tmpFile.Close()
+
+		go write(t, tmpFile.Name(), testInputs)
+
+		parent, cancel := context.WithCancel(context.Background())
+		writer := bytes.NewBuffer(nil)
+		l := log{
+			file:   tmpFile,
+			writer: writer,
+			ctx:    parent,
+			cancel: cancel,
+		}
+
+		go l.Run()
+
+		time.Sleep(intervalTime * time.Millisecond)
+		l.Stop()
+
+		expected := regexp.MustCompile(timeRegEx + " test 1\n" + timeRegEx + " test 2\n")
+		assert.Regexpf(t, expected, writer.String(), "time format is invalid")
+	})
+
+	t.Run("failure by parsing error", func(t *testing.T) {
+		tmpFile, err := ioutil.TempFile("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer tmpFile.Close()
+
+		testInvalidInputs := []string{
+			`{"t": 1581662022394, "m": "test 1", "n": 0, "s": "main"}` + "\n",
+			`{` + "\n",
```

Comment:
Could you check the log package are working after broken line was come?

Reply:
If broken line is come, logger is stopped, but build is running.